src/box/box.cc

```diff
@@ -5606,13 +5608,18 @@ box_cfg(void)
 int
 box_checkpoint(void)
 {
-	/* Signal arrived before box.cfg{} */
-	if (! is_box_configured)
-		return 0;
-
+	assert(!is_storage_shutdown);
```

Comment:
Should also check that `is_storage_initialized` is set?

Reply:
Done. Also 

- used `is_box_configured` as before the patch
- removed `is_storage_shutdown` check from `box_checkpoint` as the flag is set before iproto and client fibers shutdown

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index 0304c62d7..500694ed4 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -5608,14 +5608,14 @@ box_cfg(void)
 int
 box_checkpoint(void)
 {
-       assert(!is_storage_shutdown);
+       assert(is_box_configured);
        return gc_checkpoint();
 }

 void
 box_checkpoint_async(void)
 {
-       if (!is_storage_initialized || is_storage_shutdown)
+       if (!is_box_configured || is_storage_shutdown)
                return;
        gc_trigger_checkpoint();
 }
```